rust/lance-core/src/io/writer.rs

```diff
@@ -110,6 +111,25 @@ impl FileWriter {
     ///
     /// Returns [Err] if the schema does not match with the batch.
     pub async fn write(&mut self, batches: &[RecordBatch]) -> Result<()> {
+        let expected_schema = ArrowSchema::from(&self.schema).with_metadata(HashMap::new());
```

Comment:
Can we do it just once during the time where FileWriter is open? instead on each write()?

Reply:
Yeah good idea!